### PR TITLE
#362 kmz exporter allow user to specify output directory

### PR DESCRIPTION
**Pull request by JACochran**

### CODE REVIEW

**Review by rsupnekar — APPROVED** [review]

File: gradle.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-version=3.0.5-SNAPSHOT
+version=3.0.6-SNAPSHOT
```
Comment (stevendlander) [inline gradle.properties:1]: What version is this in regards to?  Gradle version?

File: sdk/sdk-api/src/main/java/mil/emp3/api/utils/kmz/EmpKMZExporter.java [hunk 2]
```diff
@@ -22,147 +25,354 @@ private EmpKMZExporter()
...
      */
     public static void exportToKMZ(final IMap                           map,
-                                   final IOverlay                       overlay,
                                    final boolean                        extendedData,
                                    final IEmpExportToTypeCallBack<File> callback,
                                    final String                         temporaryDirectoryLocation,
                                    final String                         kmzFileName)
     {
+        if((null == kmzFileName ) ||
+           (null == temporaryDirectoryLocation))
```
Comment (stevendlander) [inline sdk/sdk-api/src/main/java/mil/emp3/api/utils/kmz/EmpKMZExporter.java:121]: Typically we put `null` on the right side of comparisons, why the change here?

File: sdk/sdk-api/src/main/java/mil/emp3/api/utils/kmz/KMZExportThread.java [hunk 3]
```diff
@@ -213,31 +219,33 @@ public void run()
...
     }
 
-    private static void createKMZFile(final String                         kmlString,
+    /***
+     * Creates the kmz file
+     * @param kmlStringData
+     * @param temporaryDirectory
+     * @param kmzFile
+     * @param callback
+     */
```
Comment (stevendlander) [inline sdk/sdk-api/src/main/java/mil/emp3/api/utils/kmz/KMZExportThread.java:228]: Are these supposed to be filled out?

**Comment by stevendlander** [comment]
Javadoc comments dont start with a capital letter.  That is a nitpick, but it seems to be the case for all the docs i saw in this PR.

File: sdk/sdk-api/src/main/java/mil/emp3/api/utils/kmz/EmpKMZExporter.java (hunk 2 above)
Comment (JACochran) [reply sdk/sdk-api/src/main/java/mil/emp3/api/utils/kmz/EmpKMZExporter.java:121]: Matching the style in KmlExporterThread but I don't have a preference and can change it to the other way around

File: gradle.properties (hunk 1 above)
Comment (JACochran) [reply gradle.properties:1]: version of EMP

File: sdk/sdk-api/src/main/java/mil/emp3/api/utils/kmz/KMZExportThread.java (hunk 3 above)
Comment (JACochran) [reply sdk/sdk-api/src/main/java/mil/emp3/api/utils/kmz/KMZExportThread.java:228]: yup oversight
